element_array_ephys/ephys_organoids.py

```diff
@@ -128,382 +129,185 @@ class AcquisitionSoftware(dj.Lookup):
+                if "data" not in k
+            ]
+        )
 
 
 @schema
 class LFP(dj.Imported):
-    """Extracts local field potentials (LFP) from an electrophysiology recording.
-
-    Attributes:
-        EphysRecording (foreign key): EphysRecording primary key.
-        lfp_sampling_rate (float): Sampling rate for LFPs in Hz.
-        lfp_time_stamps (longblob): Time stamps with respect to the start of the recording.
-        lfp_mean (longblob): Overall mean LFP across electrodes.
-    """
-
     definition = """
-    # Acquired local field potential (LFP) from a given Ephys recording.
-    -> EphysRecording
+    -> EphysSession
     ---
-    lfp_sampling_rate: float   # (Hz)
-    lfp_time_stamps: longblob  # (s) timestamps with respect to the start of the recording (recording_timestamp)
-    lfp_mean: longblob         # (uV) mean of LFP across electrodes - shape (time,)
+    lfp_sampling_rate    : float # (Hz) down-sampled sampling rate.
+    header               : longblob
     """
 
-    class Electrode(dj.Part):
-        """Saves local field potential data for each electrode.
-
-        Attributes:
-            LFP (foreign key): LFP primary key.
-            probe.ElectrodeConfig.Electrode (foreign key): probe.ElectrodeConfig.Electrode primary key.
-            lfp (longblob): LFP recording at this electrode in microvolts.
-        """
-
+    class Trace(dj.Part):
         definition = """
         -> master
         -> probe.ElectrodeConfig.Electrode
         ---
-        lfp: longblob               # (uV) recorded lfp at this electrode
+        lfp              : blob@external-raw 
         """
 
-    # Only store LFP for every 9th channel, due to high channel density,
-    # close-by channels exhibit highly similar LFP
-    _skip_channel_counts = 9
+    @property
+    def key_source(self):
+        return EphysSession - "session_type='spike_sorting'"
 
     def make(self, key):
-        """Populates the LFP tables."""
-        acq_software = (EphysRecording * ProbeInsertion & key).fetch1("acq_software")
-
-        electrode_keys, lfp = [], []
-
-        if acq_software == "SpikeGLX":
-            spikeglx_meta_filepath = get_spikeglx_meta_filepath(key)
-            spikeglx_recording = spikeglx.SpikeGLX(spikeglx_meta_filepath.parent)
-
-            lfp_channel_ind = spikeglx_recording.lfmeta.recording_channels[
-                -1 :: -self._skip_channel_counts
-            ]
-
-            # Extract LFP data at specified channels and convert to uV
-            lfp = spikeglx_recording.lf_timeseries[
-                :, lfp_channel_ind
-            ]  # (sample x channel)
-            lfp = (
-                lfp * spikeglx_recording.get_channel_bit_volts("lf")[lfp_channel_ind]
-            ).T  # (channel x sample)
+        files = (
+            RawData
+            & f"start_time BETWEEN '{key['start_time']}' AND '{key['end_time']}'"
+        ).fetch("file_path", order_by="start_time")
+
+        TARGET_SAMPLING_RATE = 2000
```

Comment:
I suggest that we mirror the sampling rate and filtering frequencies of the Neuropixels probes:
```
The signals from each recording site are split in hardware into a spike band (30 kHz sampling rate, 500 Hz highpass filter) and an LFP band (2.5 kHz sampling rate, 1000 Hz lowpass filter). 
```
Source: [Allen Brain Observatory - Technical White Paper](https://brainmapportal-live-4cc80a57cd6e400d854-f7fdcae.divio-media.net/filer_public/80/75/8075a100-ca64-429a-b39a-569121b612b2/neuropixels_visual_coding_-_white_paper_v10.pdf)